### ISSUE TITLE
[Suggestion] Doors/Trapdoors Configurable Behaviour Mechanics

### ISSUE THREAD

**Issue by Seikikai** [PianoManu/BlockCarpentry#62]
I Want To Suggest That Like Vainilla Doors/Trapdoors That Have Special Behaviours, Frame Doors/Trapdoors Behaviours Can Be Configured Between Behave Normally, Be Opened Only By Players Or Be Opened Only By Redstone Signal. It Would Help For Keep Areas Out Of Reach For Villagers Or For Using Doors/Trapdoors As Thin Walls (A Thin Wall Block Would Be Nice Too).

![imagen](https://user-images.githubusercontent.com/54079357/137813359-2f2e85ac-6d4f-4182-8963-8525f581d3c8.png)

**Comment by PianoManu** [comment]
That's a very good idea. I somehow never saw this suggestion here in the issues section. Or maybe I read and forgot it... Thanks for this nice suggestion, I will try to implement that!

**Comment by PianoManu** [comment]
Alright, I've tested a few things and it seems to work right now. Doors/Trapdoors/Fence Gates can be locked and unlocked by right-clicking with an iron ingot (won't be consumed). They can also be changed to ignore redstone signals by right-clicking them with a piece of redstone-dust (also won't be consumed). Right-clicking them again will make them sensitive to redstone signals, again.

I guess, I've finished everything, so I'll close this issue now :)
Again, many thanks for your suggestion!
Regards, Manu
